pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -196,10 +216,14 @@ func (c *Controller) WorkloadsForWaypoint(scope model.WaypointScope) []*model.Wo
 	return res
 }
 
-// Waypoint finds all waypoint IP addresses for a given scope.  Performs first a Namespace+ServiceAccount
-// then falls back to any Namespace wide waypoints
-func (c *Controller) Waypoint(scope model.WaypointScope) []netip.Addr {
-	a := c.ambientIndex
+func (c *Controller) WorkloadsForWaypoint(scope model.WaypointScope) []*model.WorkloadInfo {
```

Comment:
nit: could just embed the `AmbientIndex` interface on `Controller` instead of giving it the `ambientIndex` name. 

Some of the names like `Lookup` and `All` get exposed though..